packages/table/src/cell/formats/truncatedFormat.tsx

```diff
@@ -74,7 +96,22 @@ export class TruncatedFormat extends React.Component<ITruncatedFormatProps, {}>
             );
         } else {
             const className = classNames(this.props.className, "bp-table-truncated-text");
-            return <div className={className}>{content}</div>;
+            return <div className={className}>{cellContent}</div>;
+        }
+    }
+
+    private shouldShowPopover(content: string) {
+        const { showPopover, truncateLength } = this.props;
+
+        switch (showPopover) {
+            case TruncatedPopoverMode.ALWAYS:
+                return true;
+            case TruncatedPopoverMode.NEVER:
```

Comment:
well we can, but i like to have all explicit enum values AND a default fallthrough